fw/rbcx-coprocessor/include/ByteFifo.hpp

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <stdint.h>
+
+template <int Size>
+class ByteFifo {
+    std::array<uint8_t, Size> m_fifo;
+    int m_head, m_tail;
+
+    int static adjust(int index, int delta) {
+        index += delta;
+        if (index >= Size) {
+            index -= Size;
+        }
+        if (index < 0) {
+            index += Size;
+        }
+        return index;
+    }
+
+public:
+    ByteFifo()
+        : m_head(0)
+        , m_tail(0) {}
+
+    uint8_t* data() const { return (uint8_t*)m_fifo.data(); }
+    size_t size() const { return m_fifo.size(); }
+
+    void setHead(int newHead) {
+        m_head = newHead;
+    }
+
+    std::pair<uint8_t*, size_t> readableRange() const {
```

Comment:
Rozumím návrhu s `readableRange` a `writeableRange` (kvůli DMA nebo navazujícímu API, které bere buffer a velikost), avšak třeba Jarka P. to na první pohled zmátlo a musel jsem s ním zapřemýšlet, proč to tak je. Uvažoval bych o dvou úpravách:
- přejmenovat na `continuousReadableRange`
- přidat komentář vysvětlující, že takto získáš místo, odkud číst, ale jak přečteš, musíš zavolat `notifyRead`.

Reply:
Je to pravda, zlepším chápetalnost.

Reply:
`readableSpan` a `writeableSpan` ? Span má definovanou spojitost

Reply:
(Btw dalo by se použít span např z GSL?)

Reply:
Vylepšeno